test/validate_v1_v2_equality.ts

```diff
@@ -1,36 +1,244 @@
-import { getCardsJson, getCardsV2Json } from "../src/index";
+import { getCardsJson, getCardSetsV2Json, getCardsV2Json, getPacksJson, getPrintingsV2Json, textToId } from "../src/index";
 import chai = require('chai');
 const expect = chai.expect;
 
-describe('card text matches between versions', () => {
+describe('Cards v1/v2 equality', () => {
   const v1Cards = getCardsJson();
-  const v1Text = new Map<string, string>();
-  const v1StrippedText = new Map<string, string>();
+  const v1CardsByTitle = new Map<string, any>();
   v1Cards.forEach(c => {
-    v1Text.set(c.title, c.text);
-    v1StrippedText.set(c.title, c.stripped_text);
+    v1CardsByTitle.set(c.title, c);
   });
+  const v2CardsByTitle = new Map<string, any>();
   const v2Text = new Map<string, string>();
   const v2StrippedText = new Map<string, string>();
+  const v2StrippedTitle = new Map<string, string>();
   const v2Cards = getCardsV2Json();
   v2Cards.forEach(c => {
+    v2CardsByTitle.set(c.title, c);
     v2Text.set(c.title, c.text);
     v2StrippedText.set(c.title, c.stripped_text);
+    v2StrippedTitle.set(c.title, c.stripped_title);
   });
 
-  it('text matches', () => {
-    expect(v1Text.size, '# of cards matches between versions').to.equal(v2Text.size);
-    v1Text.forEach((text, title) => {
-        expect(text, `Text mismatch for ${title}`).to.equal(v2Text.get(title));
+  it('correct number of cards', () => {
+    expect(v1CardsByTitle.size).to.equal(v2CardsByTitle.size);
+  });
+
+  // v2 card_type_id has {corp,runner}_identity instead of just identity.
+  it('card_type_id matches', () => {
+    v1CardsByTitle.forEach((c, title) => {
+      expect(v1CardsByTitle.get(title).type_code, `card_type_id mismatch for ${title}`)
+          .to.equal(v2CardsByTitle.get(title).card_type_id.replace(v2CardsByTitle.get(title).side_id + '_', ''));
     });
-    expect(v1StrippedText.size, '# of cards matches between versions').to.equal(v2StrippedText.size);
+  });
+
+  // v2 subtypes is a lower case list of card_subtype_ids, not the display version of subtypes.
+  it('subtypes match', () => {
+    v1CardsByTitle.forEach((c, title) => {
+      const v1 = v1CardsByTitle.get(title);
+      const v2 = v2CardsByTitle.get(title);
+      const v1Keywords = v1.keywords ? v1.keywords.split(' - ').map(x => textToId(x)) : [];
+      const v2Subtypes = v2.subtypes ? v2.subtypes : [];
+      expect(v1Keywords, `subtypes mismatch for ${title}`).to.deep.equal(v2Subtypes);
+    });
+  });
+
+  function validatev1V2Equality(fieldName, v1Getter, v2Getter) {
+    v1CardsByTitle.forEach((c, title) => {
+        expect(v1Getter(title), `${fieldName} mismatch for ${title}`).to.equal(v2Getter(title));
+    });
+  }
+
+  it('text matches', () => {
+    validatev1V2Equality('text',
+        function(title) { return v1CardsByTitle.get(title).text; }, 
```

Comment:
These could be arrow functions: `(title) => v1CardsByTitle.get(title).text`

Given that all of these are simply checking a property of an entry in the respective maps, the comparison function could take the two properties and check directly (written on a phone, apologies):
```
function checker(title, v1Prop, v2Prop) {
  expect(v1Map.get(title)[v1Prop]).to.equal(v2Map.get(title)[v2Prop]);
}
```

Reply:
ooh, nice.  That tidies things up quite a bit.   Thanks for the idiomatic JS/TS suggestions!